relay-server/src/metrics_extraction/spans/types.rs

```diff
@@ -0,0 +1,155 @@
+use std::collections::BTreeMap;
+use std::fmt::Display;
+
+use chrono::Duration;
+use relay_common::{DurationUnit, MetricUnit, UnixTimestamp};
+use relay_metrics::{Metric, MetricNamespace, MetricValue};
+
+use crate::metrics_extraction::IntoMetric;
+
+#[derive(Clone, Debug)]
+pub(crate) enum SpanMetric {
+    User {
+        value: String,
+        tags: BTreeMap<SpanTagKey, String>,
+    },
+    Duration {
+        value: Duration,
+        tags: BTreeMap<SpanTagKey, String>,
+    },
+    ExclusiveTime {
+        value: f64,
+        tags: BTreeMap<SpanTagKey, String>,
+    },
+}
```

Comment:
Just a thought.. would it make sense to do like this, since `tags` is shared ?

``` Rust
enum SpanMetricType{
  User: String,
  Duration: Duration,
  ExclusiveTime: f64,
}

struct SpanMetric{
  ty: SpanMetricType,
  tags: BTreeMap<SpanTagKey, String>
}

Reply:
Although this may work, I think the API of the current approach is simpler: one object vs two (in your suggestion). See how you'd interact with it:

```rs
// Current approach
SpanMetric::User {
    value: user_tag,
    tags: span_tags.clone()
}
.into_metric(timestamp);

// Suggested approach
SpanMetric {
    ty: SpanMetricType::User(user_tag),
    tags: span_tags.clone()
}
.into_metric(timestamp);
```

If/when we increase the number of metrics, we can revisit this and it may make sense at that point. I believe this simpler approach is better suited for now.